biz.aQute.bnd.maven/src/aQute/bnd/maven/lib/resolve/BndrunContainer.java

```diff
@@ -323,12 +323,13 @@ public void setEEfromBuild(Run run) {
 						return property;
 					}
 					// so fallback to the currently running Java version
-					return System.getProperty("java.version");
+					return System.getProperty("java.vm.specification.version");
 				})
 				.flatMap(EE::highestFromTargetVersion)
-				// if that all fails at least we know bnd needs at least
-				// Java 8 at this point
-				.orElse(EE.JavaSE_1_8);
+				.orElseGet(() -> EE.highestFromTargetVersion(System.getProperty("java.vm.specification.version"))
```

Comment:
I think this should be `java.specification.version` rather than `java.vm.specification.version`.

There is a minuscule possibility that the System.getProperty call could return null and then the EE.highestFromTargetVersion call will NPE. So this should probably be:

```java
.orElseGet(() -> Optional.ofNullable(System.getProperty("java.specification.version"))
    .flatMap(EE::highestFromTargetVersion)
    // if that all fails at least we know bnd needs at least
    // Java 8 at this point
    .orElse(EE.JavaSE_1_8));
```

Reply:
done!